src/protections/ProtectionSettings.ts

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2021 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+export class ProtectionSettingValidationError extends Error {};
```

Comment:
Does this work? I have vague recollections that subclassing `Error` properly was a nightmare in JavaScript, but this may have changed with more recent versions.

Reply:
"work" in what sense? there's an integration test that checks it ends up thrown and that test passes

Reply:
Well, maybe a wart of JavaScript was finally fixed :)